identity-service/src/utils/withdrawUSDCInstructionsHelpers.js

```diff
@@ -0,0 +1,68 @@
+const {
+  ASSOCIATED_TOKEN_PROGRAM_ID,
+  TOKEN_PROGRAM_ID
+} = require('@solana/spl-token')
+const { Keypair } = require('@solana/web3.js')
+const config = require('../config')
+
+const solanaClaimableTokenProgramAddress = config.get(
+  'solanaClaimableTokenProgramAddress'
+)
+
+const allowedProgramIds = [
+  solanaClaimableTokenProgramAddress,
+  /* secp */ 'KeccakSecp256k11111111111111111111111111111'
+]
+
+const isRelayAllowedProgram = (instructions) => {
+  for (const instruction of instructions) {
+    if (!allowedProgramIds.includes(instruction.programId)) {
+      return false
+    }
+  }
+  return true
+}
+
+const checkCreateAccountInstruction = (instruction) => {
+  const isCreateInstruction =
+    instruction.programId === ASSOCIATED_TOKEN_PROGRAM_ID.toString()
+  return isCreateInstruction
+}
+
+const checkCloseAccountInstruction = (instruction) => {
+  const feePayerKeypairs = config.get('solanaFeePayerWallets')
+  const feePayerPubkeys = feePayerKeypairs.map((wallet) =>
+    Keypair.fromSecretKey(
+      Uint8Array.from(wallet.privateKey)
+    ).publicKey.toString()
+  )
+  const isCloseInstruction =
+    instruction.programId === TOKEN_PROGRAM_ID.toString() &&
+    instruction.data &&
+    instruction.data.data &&
+    instruction.data.data[0] === 9
+  const isFeePayerReimbursed = feePayerPubkeys.includes(
+    instruction.keys[1].pubkey
+  )
+  return isCloseInstruction && isFeePayerReimbursed
+}
+
+/**
+ * Checks if the given instructions are a USDC withdrawal swap transaction. This is a special case
+ * where a USDC associated token account is created, used for swapping USDC for SOL, and then closed.
+ */
+const isUSDCWithdrawalTransaction = (instructions) => {
```

Comment:
_Might_ be worth making a more generally usable check for `createAccount` instructions that checks that for each account that's created, that same account gets closed and our feepayer reimbursed, rather than checking for explicitly the first instruction is a create and the last instruction is a (notably potentially unrelated) close.

I don't think there's enough of a use case for us using it elsewhere right now though so this is fine.

Reply:
hm you're right. we should explicitly check that they are the same accounts. will follow up with that.